scripts/build_binary_and_layer.sh

```diff
@@ -35,19 +43,16 @@ TARGET_DIR=$(pwd)/$EXTENSION_DIR
 echo "Building Lambda extension binary"
 cd $SERVERLESS_CMD_PATH
 
-LD_FLAGS=""
+# Add the current version number to the tags package before compilation
+LD_FLAGS="-X github.com/DataDog/datadog-agent/pkg/serverless/tags.currentExtensionVersion=${VERSION}"
 
-if [ "$COMPRESS" = true ]; then
-    LD_FLAGS="-s -w"
+if [ "$PRODUCTION" = true ]; then
+    LD_FLAGS+=" -s -w"
```

Comment:
I think we should always build with those flags in order to have prod and non-pro very similar